src/app/file-upload-modal/types.ts

```diff
@@ -0,0 +1,6 @@
+import { FormControl } from "@angular/forms";
+
+export interface UploadedFile extends File {
+  id?: number | string;
```

Comment:
just curious why this can be both a number or string.. should it just be one, or is there some backwards compatibility think you are checking, or is it a state thing ie props vs code, I'm not an expert here, but i kind of feel you want to make these either a string or a number if at all possible.

Reply:
For context, id is used for two things in the File Upload Modal Component.

- When files are uploaded, they're given a pseudo-random ID which is purely a number. This is so they can't be added twice to the list of selected files(let's say a user is allowed to select 3). 
- Right now, the file upload modal only allows you to upload files(giving them each number IDs) but in the future, previously-saved files will be loaded from the API and displayed here. Those files will use the mongo ObjectID, which ends up in the app as a string.

So, basically, I need a way to keep track of selected files for those that have been uploaded. I didn't want to import a library just to generate a random ID that's a string.